src/filesystem/interface.js

```diff
@@ -262,94 +263,98 @@ function FileSystem(options, callback) {
       impl.ensureRootDirectory(context, complete);
     }
   });
-}
-
-// Expose storage providers on FileSystem constructor
-FileSystem.providers = providers;
-
-/**
+  FileSystem.prototype.promises = {};
+  /**
  * Public API for FileSystem
  */
-[
-  'open',
-  'chmod',
-  'fchmod',
-  'chown',
-  'fchown',
-  'close',
-  'mknod',
-  'mkdir',
-  'rmdir',
-  'stat',
-  'fstat',
-  'link',
-  'unlink',
-  'read',
-  'readFile',
-  'write',
-  'writeFile',
-  'appendFile',
-  'exists',
-  'lseek',
-  'readdir',
-  'rename',
-  'readlink',
-  'symlink',
-  'lstat',
-  'truncate',
-  'ftruncate',
-  'utimes',
-  'futimes',
-  'setxattr',
-  'getxattr',
-  'fsetxattr',
-  'fgetxattr',
-  'removexattr',
-  'fremovexattr'
-].forEach(function(methodName) {
-  FileSystem.prototype[methodName] = function() {
-    var fs = this;
-    var args = Array.prototype.slice.call(arguments, 0);
-    var lastArgIndex = args.length - 1;
-
-    // We may or may not get a callback, and since node.js supports
-    // fire-and-forget style fs operations, we have to dance a bit here.
-    var missingCallback = typeof args[lastArgIndex] !== 'function';
-    var callback = maybeCallback(args[lastArgIndex]);
-
-    var error = fs.queueOrRun(function() {
-      var context = fs.provider.openReadWriteContext();
-
-      // Fail early if the filesystem is in an error state (e.g.,
-      // provider failed to open.
-      if(FS_ERROR === fs.readyState) {
-        var err = new Errors.EFILESYSTEMERROR('filesystem unavailable, operation canceled');
-        return callback.call(fs, err);
-      }
+  [
```

Comment:
Are these tabs?  Can we not do all these unnecessary whitespace changes so the history isn't affected for things unrelated to your fix?

Reply:
This is done because I want to bind `this` reference to `promises` object in the same loop that regular functions are attached to the prototype. For that, I need to put the loop **inside** the constructor, this is where extra spaces are coming from. Previously the loop was in the global scope, now it is inside the function.